vim25/types/types.go

```diff
@@ -48377,7 +48377,7 @@ type VirtualMachineConfigInfo struct {
 	VAppConfig                   BaseVmConfigInfo                           `xml:"vAppConfig,omitempty,typeattr"`
 	VAssertsEnabled              *bool                                      `xml:"vAssertsEnabled"`
 	ChangeTrackingEnabled        *bool                                      `xml:"changeTrackingEnabled"`
-	Firmware                     string                                     `xml:"firmware,omitempty"`
+	Firmware                     GuestOsDescriptorFirmwareType              `xml:"firmware,omitempty"`
```

Comment:
The code in vim25/types is generated, you don't want to modify them manually.  The type also needs to be a `string` in this case, must be same as the xsi type.